src/DssSpell.t.sol

```diff
@@ -8,89 +8,79 @@ import "dss-interfaces/Interfaces.sol";
+        uint256 prevDaiSchuppi       = dai.balanceOf(ULTRASCHUPPI);
+        uint256 prevDaiFeedblack     = dai.balanceOf(FEEDBLACKLOOPS);
+        uint256 prevDaiMakerMan      = dai.balanceOf(MAKERMAN);
+        uint256 prevDaiAcreInvest    = dai.balanceOf(ACREINVEST);
+        uint256 prevDaiJustinCase    = dai.balanceOf(JUSTINCASE);
+        uint256 prevDaiGfxLabs       = dai.balanceOf(GFXLABS);
+
+
+        uint256 amountTotal     = amountFlipFlop + amountSchuppi + amountFeedblack
+                                + amountMakerMan + amountAcreInvest + amountJustinCase
+                                + amountGfxLabs;
+
+        assertEq(amountTotal, 48_290);
+
+        assertEq(vat.can(address(pauseProxy), address(daiJoin)), 1);
 
-    struct Award {
-        address usr;
-        uint48  bgn;
-        uint48  clf;
-        uint48  fin;
-        address mgr;
-        uint8   res;
-        uint128 tot;
-        uint128 rxd;
+        vote(address(spell));
+        spell.schedule();
+        hevm.warp(spell.nextCastTime());
+        spell.cast();
+        assertTrue(spell.done());
+
+        assertEq(vat.can(address(pauseProxy), address(daiJoin)), 1);
+
+        assertEq(vat.sin(address(vow)) - prevSin, amountTotal * RAD);
+        assertEq(dai.balanceOf(FLIPFLOPFLAP) - prevDaiFlipFlop, amountFlipFlop * WAD);
+        assertEq(dai.balanceOf(ULTRASCHUPPI) - prevDaiSchuppi, amountSchuppi * WAD);
+        assertEq(dai.balanceOf(FEEDBLACKLOOPS) - prevDaiFeedblack, amountFeedblack * WAD);
+        assertEq(dai.balanceOf(MAKERMAN) - prevDaiMakerMan, amountMakerMan * WAD);
+        assertEq(dai.balanceOf(ACREINVEST) - prevDaiAcreInvest, amountAcreInvest * WAD);
+        assertEq(dai.balanceOf(JUSTINCASE) - prevDaiJustinCase, amountJustinCase * WAD);
+        assertEq(dai.balanceOf(GFXLABS) - prevDaiGfxLabs, amountGfxLabs * WAD);
     }
 
-    mapping (uint256 => Award) internal awards;
-
-    function testVestTransfer() public {
-
-        uint256 prevTotalVest = 0;
-
-        for(uint256 i = 1; i <= VestAbstract(addr.addr("MCD_VEST_MKR")).ids(); i++) {
-            assertTrue(VestAbstract(addr.addr("MCD_VEST_MKR")).valid(i));
-            (
-                address usr,
-                uint48 bgn,
-                uint48 clf,
-                uint48 fin,
-                address mgr,
-                uint8 res,
-                uint128 tot,
-                uint128 rxd
-            ) = VestAbstract(addr.addr("MCD_VEST_MKR")).awards(i);
-            awards[i] = Award({
-                usr: usr,
-                bgn: bgn,
-                clf: clf,
-                fin: fin,
-                mgr: mgr,
-                res: res,
-                tot: tot,
-                rxd: rxd
-            });
-            prevTotalVest += tot;
-        }
+    function testOptimismRecovery() public {
+        uint256 amountLost = 10 * MILLION;
+        address l1Escrow     = 0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65;
```

Comment:
put it in addresses_mainnet ?

Reply:
already there apparently, i'll use that one